tests/test_subpixmaxima.py

```diff
@@ -10,145 +12,113 @@
-    maxima = maxima.squeeze(0)
-    assert maxima.shape == example["keypoints"].shape
-    assert maxima.shape[0] // 2 == confidence.shape[1]
 
-    # remove model/data from gpu; then cache can be cleared
-    del example
-    del maxima, confidence
-    torch.cuda.empty_cache()  # remove tensors from gpu
+    # find soft argmax and confidence of ground truth heatmap
+    softmaxes_gt = spatial_softmax2d(
+        heatmap_gt.to(_TORCH_DEVICE),
+        temperature=torch.tensor(100).to(_TORCH_DEVICE))
+    preds_gt = spatial_expectation2d(softmaxes_gt, normalized_coordinates=False)
+    confidences_gt = torch.amax(softmaxes_gt, dim=(2, 3))
 
-    dl = DataLoader(dataset, batch_size=2)
-    batch = next(iter(dl))
-    del dataset
-    del dl
-    torch.cuda.empty_cache()  # remove tensors from gpu
+    # find soft argmax and confidence of generated heatmap
+    softmaxes_torch = spatial_softmax2d(
+        heatmap_torch.to(_TORCH_DEVICE),
+        temperature=torch.tensor(100).to(_TORCH_DEVICE))
+    preds_torch = spatial_expectation2d(softmaxes_torch, normalized_coordinates=False)
+    confidences_torch = torch.amax(softmaxes_torch, dim=(2, 3))
 
-    maxima1, confidence1 = SubPixMax.run(batch["heatmaps"].to(_TORCH_DEVICE))
-    print(maxima1.shape, confidence1.shape)
-
-    # remove model/data from gpu; then cache can be cleared
-    del batch
-    del maxima1, confidence1
-    torch.cuda.empty_cache()  # remove tensors from gpu
-
-
-def test_generate_keypoints():
-    data_transform = []
-    data_transform.append(
-        iaa.Resize({"height": 384, "width": 384})
-    )  # dlc dimensions need to be repeatably divisable by 2
-    imgaug_transform = iaa.Sequential(data_transform)
-    dataset = HeatmapDataset(
-        root_directory="toy_datasets/toymouseRunningData",
-        csv_path="CollectedData_.csv",
-        header_rows=[1, 2],
-        imgaug_transform=imgaug_transform,
-    )
-    example = dataset.__getitem__(idx=0)
-    assert example["keypoints"].shape == (torch.Size([34]))
-    gt_heatmap = example["heatmaps"].unsqueeze(0)
-    gt_keypts = example["keypoints"].unsqueeze(0).reshape(1, 17, 2)
-    assert gt_heatmap.shape == (1, 17, 96, 96)
-    torch_heatmap = generate_heatmaps(
-        gt_keypts, height=384, width=384, output_shape=(96, 96)
-    )
-    SubPixMax = SubPixelMaxima(
-        output_shape=(96, 96),
-        output_sigma=torch.tensor(1.25, device=_TORCH_DEVICE),
-        upsample_factor=torch.tensor(2, device=_TORCH_DEVICE),
-        coordinate_scale=torch.tensor(4, device=_TORCH_DEVICE),  # 2 ** 2
-        confidence_scale=torch.tensor(1, device=_TORCH_DEVICE),
-        threshold=None,
-        device=_TORCH_DEVICE,
-    )
-    og_maxima, gt_confidence = SubPixMax.run(gt_heatmap.to(_TORCH_DEVICE))
-    torch_maxima, confidence_t = SubPixMax.run(torch_heatmap.to(_TORCH_DEVICE))
-    assert (og_maxima == torch_maxima).all()
-    assert (gt_confidence == confidence_t).all()
+    assert(preds_gt == preds_torch).all()
```

Comment:
seems like it could be a strong test, but if I understand correctly `example_heatmaps` and `heatmap_torch` are by definition the same thing. so we're not really testing the subpix maxima no? 
the stronger test is to compare the 2D gt keypoints and the 2d predicted keypoints after subpixel maxima. they may not be really equal, but we can accept some `tolerance`. 
Using `render_2d_gaussian` from `kornia` will get you closer to the the equality I think

Reply:
if i understand correctly this test is actually for `generate_heatmaps` (despite its name) so we can just scrap altogether if we replace with kornia functions